raiden/channel.py

```diff
@@ -622,53 +620,27 @@ def locked(self):
     def outstanding(self):
         return self.our_state.locked()
 
-    def channel_closed(self, block_number):
-        self.external_state.register_block_alarm(self.blockalarm_for_settle)
-
+    def channel_closed(self, block_number):  # pylint: disable=unused-argument
         balance_proof = self.our_state.balance_proof
-
         transfer = balance_proof.transfer
-        unlock_proofs = balance_proof.get_known_unlocks()
-
-        self.external_state.unlock(self.our_state.address, unlock_proofs)
 
-        # check the published messages for the correct transferred_amount
-        # and dispute if there is a discrepancy
         our_closed_transferred = self.external_state.query_transferred_amount(
             self.our_state.address
         )
         partner_closed_transferred = self.external_state.query_transferred_amount(
             self.partner_state.address
         )
-        if (
-                self.our_state.transferred_amount != our_closed_transferred or
-                self.partner_state.transferred_amount != partner_closed_transferred
-        ):
-            self.external_state.update_transfer(self.our_state.address, transfer)
-
-    def blockalarm_for_settle(self, block_number):
-        def _settle():
-            # do not call settle if already settled, the event polling
-            # might be lagging behind.
-            settled_block = self.external_state.query_settled()
-            if settled_block != 0:
-                self.external_state.set_settled(settled_block)
-                log.info('channel automatically settled')
-                return
-
-            try:
-                self.external_state.settle()
-            except:
-                log.exception('Timedout while calling settle')
 
-        closed = self.external_state.closed_block != 0
-        after_settle_timeout = block_number >= (
-            self.external_state.closed_block + self.settle_timeout
+        need_to_dispute = (
+            self.our_state.transferred_amount != our_closed_transferred or
+            self.partner_state.transferred_amount != partner_closed_transferred
```

Comment:
Humm, actually the courtesy transfer was removed, so we always need to call updateTransfer.